lib/vtls/schannel.c

```diff
@@ -672,12 +691,13 @@ schannel_connect_step2(struct connectdata *conn, int sockindex)
     infof(data, "schannel: SSL/TLS handshake complete\n");
   }
 
-#ifdef _WIN32_WCE
-  /* Windows CE doesn't do any server certificate validation.
-     We have to do it manually. */
-  if(conn->ssl_config.verifypeer)
+  if(conn->ssl_config.verifypeer &&
+     (DEFAULT_MANUAL_CRED_VALIDATION ||
+     (conn->ssl_config.CAfile && CAFILE_SUPPORTED))) {
```

Comment:
Instead of checking these conditions here again, I would advise to use a variable to re-use the result of the checks in https://github.com/curl/curl/pull/1325/files#diff-2ba9d5898aad590d89cd83e17ad244fbL210. Or maybe just check if schannel_cred.dwFlags contains SCH_CRED_MANUAL_CRED_VALIDATION and conn->ssl_config.verifypeer is set.